app/models/katello/host_collection.rb

```diff
@@ -30,8 +30,13 @@ class HostCollection < Katello::Model
     scoped_search :on => :organization_id, :complete_value => true
 
     def validate_max_content_hosts
+      if new_record?
```

Comment:
You can move this check directly to the validation, `validates ... :if => ->(host_collection) { host_collection.new_record? }`